pkg/types/types.go

```diff
@@ -23,67 +23,136 @@ import (
 	"github.com/project-oak/transparent-release/pkg/intoto"
 )
 
-// ValidatedProvenance wraps an intoto.Statement representing a valid SLSA provenance statement.
-// A provenance statement is valid if it contains a single subject, with a SHA256 hash.
-type ValidatedProvenance struct {
+// ProvenanceIR wraps an intoto.Statement representing a valid SLSA provenance statement. A provenance statement is valid if it contains a single subject, with a SHA256 hash.
+// ProvenanceIR also holds internal intermediate representations of data from provenances. We want to map different provenances of different build types to ProvenanceIR, so
+// all fields except for `provenanceStatement` are optional.
+type ProvenanceIR struct {
```

Comment:
I don't see why this is moved from `common` to here. In particular, `ProvenanceIR` is an internal representation, and belongs to `internal` not `pkg`, which by convention provides the public interface. 

Please move the content of this file to `common`, and leave only `ParseStatementData` in this file.